mentat/include_files.py

```diff
@@ -44,104 +23,224 @@ def is_file_text_encoded(abs_path: Path):
         return False
 
 
-def abs_files_from_list(paths: list[Path], check_for_text: bool = True):
```

Comment:
Why do we need to rewrite all of this logic? As far as I'm aware the only git related part of include_files was the git ignore directory logic

Reply:
See https://github.com/AbanteAI/mentat/pull/286#issuecomment-1821996841